plugins/verification/src/main/java/org/qbicc/plugin/verification/MemberResolvingBasicBlockBuilder.java

```diff
@@ -237,6 +230,8 @@ public Value checkcast(final Value value, final TypeDescriptor desc) {
             // we can't just bitcast it really, in fact it's an error unless the actual value is zero
             if (value instanceof Literal && ((Literal) value).isZero()) {
                 return ctxt.getLiteralFactory().zeroInitializerLiteralOfType(castType);
+            } else if (value instanceof Load) {
+                return value;
```

Comment:
This wouldn't be correct either, since it's possible to insert a definitely-incorrect cast on the result of a load. I think what we probably want to do here is add a check to see if the `castType` is equal to the value type being cast (or is a generic cast type like `CNative.object`), and if so, pass the value through.  Would that be enough, or was there another error you've observed here?

Reply:
Yes that works.